libnest/src/package/requirement.rs

```diff
@@ -219,6 +217,213 @@ impl std::fmt::Display for PackageRequirement {
     }
 }
 
+/// A structure representing a package requirement: parts of a package name and a
+/// version requirement.
+///
+/// Each part may be optional except the category and the package name (you can match,
+/// for exemple, any version of a package named 'sys-bin/gcc' in any repository)
+///
+/// The version requirement follows SemVer v2
+#[derive(Serialize, Deserialize, Clone, Eq, PartialEq, Ord, PartialOrd, Hash, Debug)]
+pub struct PackageRequirement {
+    repository: Option<RepositoryName>,
+    category: CategoryName,
+    name: PackageName,
+    version_requirement: VersionReq,
+}
+
+impl PackageRequirement {
+    /// Creates a package requirement that matches the given [`PackageFullName`] and version requirement
+    #[inline]
+    pub fn from(full_name: &PackageFullName, version_req: VersionReq) -> PackageRequirement {
+        PackageRequirement {
+            repository: Some(full_name.repository().clone()),
+            category: full_name.category().clone(),
+            name: full_name.name().clone(),
+            version_requirement: version_req,
+        }
+    }
+
+    /// Creates a package requirement that matches the given [`PackageFullName`] and version requirement.
+    #[inline]
+    pub fn from_id(id: &PackageID) -> PackageRequirement {
```

Comment:
This somehow wasn't implemented as a `From<PackageID>` for `SoftPackageRequirement`, I opened an issue for it: see #100